libpromises/evalfunction.c

```diff
@@ -2931,7 +2931,10 @@ JsonElement *DefaultTemplateData(const EvalContext *ctx, const char *wantbundle)
             if (NULL != scope_obj)
             {
                 char *lval_key = VarRefToString(var->ref, false);
-                JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                if (NULL == strchr(lval_key, '#')) // don't collect mangled refs
+                {
+                    JsonObjectAppendElement(scope_obj, lval_key, RvalToJson(var->rval));
+                }
```

Comment:
Haha, you're right, I wish I hadn't. :-)
I suppose this is appropriate then, since the data for the bundle is not supposed to include any variables from outside the bundle. The ``this#promise_filename`` is an exception merely because it is in fact local to that bundle.